sesman/startwm.sh

```diff
@@ -55,17 +55,38 @@ post_start()
   return 0
 }
 
+get_xdg_session_startupcmd()
+{
+  # DESKTOP_SESSION should be set in sesman.ini in the SessionVariables section.
+  # If set and valid then the STARTUP command will be taken from there
+  if [ -n "$1" ] && [ -d /usr/share/xsessions ] \
+    && [ -f "/usr/share/xsessions/$1.desktop" ]; then
+    STARTUP=$(grep ^Exec= "/usr/share/xsessions/$1.desktop")
+    STARTUP=${STARTUP#Exec=*}
+    XDG_CURRENT_DESKTOP=$(grep ^DesktopNames= "/usr/share/xsessions/$1.desktop")
+    XDG_CURRENT_DESKTOP=${XDG_CURRENT_DESKTOP#DesktopNames=*}
+    echo "env XDG_CURRENT_DESKTOP=$XDG_CURRENT_DESKTOP $STARTUP"
+  fi
+}
+
 #start the window manager
 wm_start()
 {
   if [ -r /etc/default/locale ]; then
     . /etc/default/locale
     export LANG LANGUAGE
   fi
-
+  
   # debian
   if [ -r /etc/X11/Xsession ]; then
     pre_start
+    
+    # STARTUP is the default startup command.
+    # if $1 is empty and STARTUP was not set
+    # /etc/X11/Xsession.d/50x11-common_determine-startup will fallback to
+    # x-session-manager
+    STARTUP="$(get_xdg_session_startupcmd "$DESKTOP_SESSION")"
```

Comment:
Small point, but you can avoid a sub-process invocation for the common case (i.e. DESKTOP_SESSION not set) by testing DESKTOP_SESSION here. Also, it's possible (??) that STARTUP may be set by the user. How about:-

```sh
if [ -z "$STARTUP" ] && [ -n "$DESKTOP_SESSION" ]; then
    STARTUP="$(get_xdg_session_startupcmd "$DESKTOP_SESSION")"
fi
```

I'm making this up as I go along - please come back at me if you disagree.

Reply:
sounds reasonable. if we don't do that we may create a regression if one used the STARTUP "trick" in .profile

Reply:
Good point about the regression. That hadn't occurred to me.